### PR TITLE
Allow passing of parameters to reactive sql client connect options

### LINKED ISSUE #25914: Postgres reactive configuration doesn't allow for arbitrary parameters on connect string
### Describe the bug

I'm trying to use Hibernate Reactive with Panache to connect the [quarkus superheroes `rest-heroes` app](https://github.com/quarkusio/quarkus-super-heroes/tree/main/rest-heroes) to CockroachDB. The app (Quarkus 2.9.2) works fine with Dev Services or a postgres db I stand up on my own.

I've done a similar thing with the [quarkus superheroes `rest-villains` app](https://github.com/quarkusio/quarkus-super-heroes/tree/main/rest-villains) with JDBC & Hibernate ORM with Panache and everything works fine.

When connecting to the Cockroach DB DBaaS, it is [mandatory that you include an `options=--clister=<cluster_name>` parameter](https://www.cockroachlabs.com/docs/v22.1/connection-parameters#connect-using-a-url), so I tried this configuration in the `application.yml`:

```yaml
quarkus:
  datasource:
    username: myusername
    password: mypassword
    reactive:
      url: postgresql://free-tier14.aws-us-east-1.cockroachlabs.cloud:26257/defaultdb?sslmode=require&options=--cluster%3Dheroes-db-2463
      postgresql:
        ssl-mode: require
```

### Expected behavior

I would expect the `options` connection param to be passed along to the PG driver.

### Actual behavior

Instead I get this error:

```
15:48:28 INFO  [or.hi.re.pr.im.ReactiveIntegrator] (JPA Startup Thread: default-reactive) HR000001: Hibernate Reactive

15:48:29 WARN  [or.hi.re.pr.se.ReactiveGenerationTarget] (vert.x-eventloop-thread-0) HR000021: DDL command failed [io.vertx.pgclient.PgException: FATAL: codeParamsRoutingFailed: missing cluster identifier (08004)]
15:48:29 WARN  [or.hi.re.pr.se.ReactiveGenerationTarget] (vert.x-eventloop-thread-0) HR000021: DDL command failed [io.vertx.pgclient.PgException: FATAL: codeParamsRoutingFailed: missing cluster identifier (08004)]
```

This indicated that the parameter was not included.

I did this as a test and it works fine:

```java
@ApplicationScoped
public class StartupHelper {
  private final Vertx vertx;

  public StartupHelper(Vertx vertx) {
    this.vertx = vertx;
  }

  public void config(@Observes StartupEvent startupEvent) {
    Log.infof("Inside %s.config", getClass().getName());

    var connectOptions = new PgConnectOptions()
      .setPort(26257)
      .setHost("free-tier14.aws-us-east-1.cockroachlabs.cloud")
      .setDatabase("defaultdb")
      .setUser("myusername")
      .setPassword("mypassword")
      .setSslMode(SslMode.REQUIRE)
      .addProperty("options", "--cluster=heroes-db-2463");

    var poolOptions = new PoolOptions()
      .setMaxSize(5);

    var client = PgPool.client(this.vertx, connectOptions, poolOptions);

    client.query("DROP TABLE IF EXISTS Hero").execute()
      .onItem().invoke(() -> Log.info("Dropped Hero table"))
      .flatMap(r -> client.query("DROP SEQUENCE IF EXISTS hibernate_sequence").execute()
        .onItem().invoke(() -> Log.info("Dropped hibernate_sequence"))
      )
      .flatMap(r -> client.query("CREATE SEQUENCE hibernate_sequence START 1 INCREMENT 1").execute()
        .onItem().invoke(() -> Log.info("Created hibernate_sequence"))
      )
      .flatMap(r -> client.query("CREATE TABLE Hero (id int8 NOT NULL, level int4 NOT NULL, name VARCHAR(50) NOT NULL, otherName VARCHAR(255), picture VARCHAR(255), powers TEXT, PRIMARY KEY (id))").execute()
        .onItem().invoke(() -> Log.info("Created Hero table"))
      )
      .flatMap(r -> client.query("INSERT INTO hero(id, name, otherName, picture, powers, level) VALUES (nextval('hibernate_sequence'), 'Chewbacca', '', 'https://raw.githubusercontent.com/quarkusio/quarkus-super-heroes/characterdata/images/chewbacca--684239239428094811.jpg', 'Super Strength, Agility, Animal Attributes, Jaw Strength, Longevity, Marksmanship, Weapons Master', 30)").execute()
        .onItem().invoke(() -> Log.info("Inserted record into Hero table"))
      )
      .flatMap(r -> client.query("SELECT * FROM hero").execute()
        .onItem().invoke(ar -> Log.infof("Got %d rows", ar.size()))
      )
      .await().atMost(Duration.ofSeconds(20));

    client.close();
  }
}
```

Now the question is - how do I get that `options` parameter to stick in the Quarkus configuration? Is it a problem in the upstream Vert.x pg client? Or is there something we can/should do in Quarkus to solve for this? Maybe adding a config param in `io.quarkus.reactive.pg.client.runtime.DataSourceReactivePostgreSQLConfig` and read that in `io.quarkus.reactive.pg.client.runtime.PgPoolRecorder`?

### How to Reproduce?

Steps to reproduce:

1. You will need a cockroach labs account. I have a DB set up with credentials I can share if you would like. Please ping me privately (I don't want to share publicly).
2. Clone https://github.com/quarkusio/quarkus-super-heroes
3. `cd` into `rest-heroes`
4. Add the following to `src/main/resources/application.yml`:
   ```yaml
   quarkus:  
     datasource:
       username: myusername 
       password: mypassword
       reactive:
        url: postgresql://free-tier14.aws-us-east-1.cockroachlabs.cloud:26257/defaultdb?sslmode=require&options=--cluster%3Dheroes-db-2463
        postgresql:
          ssl-mode: require
   ```
5. Run `quarkus dev`

### Output of `uname -a` or `ver`

```
Darwin edeandrea-m1pro 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
```

### Output of `java -version`

```
openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode)
```

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.9.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.3-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.4", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by edeandrea**
Allow passing of parameters to reactive sql client connect options

Fixes #25914

### CODE REVIEW

**Comment by edeandrea** [comment]
@geoand / @Sanne I took a stab at this. I implemented it using a generic `properties` container that gets passed along to the underlying `SqlConnectOptions.addProperty`.  This way, configuration like this works:

```yaml
quarkus:
  datasource:
    username: myusername
    password: mypassword
    reactive:
      url: postgresql://free-tier14.aws-us-east-1.cockroachlabs.cloud:26257/defaultdb
      properties:
        options: --cluster=heroes-db-2463
      postgresql:
        ssl-mode: require
```

If you'd rather just have a specific config property for `options` rather than a `Map` for `properties`, I can change things around.

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java [hunk 1]
```diff
@@ -145,4 +146,10 @@ public class DataSourceReactiveRuntimeConfig {
      */
     @ConfigItem
     public Optional<String> name;
+
+    /**
+     * Additional properties for the client, which will be sent to server at the connection start.
+     */
+    @ConfigItem
+    public Map<String, String> properties;
```
Comment (tsegismont) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:154]: Perhaps, for consistency, we could call this `additionalProperties`:

https://github.com/quarkusio/quarkus/blob/15677f05c40f6257922e55610321baadfe9c8a9c/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java#L135-L139

**Review by tsegismont — APPROVED** [review]
Thank you @edeandrea 

Just one suggestion, not mandatory.

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java (hunk 1 above)
Comment (edeandrea) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:154]: Sure I can do that. I was staying consistent with the `getProperties`/`setProperties`/`addProperty` on `io.vertx.sqlclient.SqlConnectOptions`

Comment (tsegismont) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java]: Thank you

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java [hunk 2]
```diff
@@ -145,4 +146,10 @@ public class DataSourceReactiveRuntimeConfig {
      */
     @ConfigItem
     public Optional<String> name;
+
+    /**
+     * Additional properties for the client, which will be sent to server at the connection start.
```
Comment (tsegismont) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:151]:
```suggestion
     * Other unspecified properties to be passed to the Reactive SQL Client when creating new connections.
```

**Comment by kdubb** [comment]
@edeandrea Just by reading, the naming is a little confusing. Does cockroach support a `PGConnectOptions` option named `options`?

We need to set the `search_path`, would I do that as follows?

```properties
quarkus.datasource.username=myusername
quarkus.datasource.password=mypassword
quarkus.datasource.reactive.url=postgresql://host/db
quarkus.datasource.reactive.additional-properties.search_path=my_schema,public
```

**Comment by kdubb** [comment]
BTW the naming issues start with vertx. As you know the name of the class is `PGConnectOptions` but it has `addProperty` and `addProperties`. So, naming it `additionalProperties` also seems inline as long as we ignore the "options" outlier.

**Comment by edeandrea** [comment]
@kdubb CockroachDB server less DBaaS requires a connection parameter names `options` which has a value of `--cluster=<cluster_namr>`. See their documentation (https://www.cockroachlabs.com/docs/v22.1/connection-parameters#connect-using-a-url).

With this change, `quarkus.datasource.reactive.additional-properties.search_path=my_schema,public` would set

`search_path=my_schema,public` as a property on the `SqlConnectOptions` (the parent class of `PgConnectOptions`). This change flows on all the reactive drivers, not just the postgres one.

**Comment by kdubb** [comment]
@edeandrea Perfect. Thanks this is exactly what we need.

@gsmet This seems ready to go. Who do we ping to get this in for 2.10? I don't think I'm allowed to exercise my labeling powers to add this to the milestone lol

**Comment by geoand** [comment]
> Who do we ping to get this in for 2.10? I don't think I'm allowed to exercise my labeling powers to add this to the milestone lol

As long as it's merged in the next few days, it will be part of `2.10`. 

@Sanne can you take a look at it?

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java [hunk 3]
```diff
@@ -145,4 +146,10 @@ public class DataSourceReactiveRuntimeConfig {
      */
     @ConfigItem
     public Optional<String> name;
+
+    /**
+     * Other unspecified properties to be passed to the Reactive SQL Client when creating new connections.
```
Comment (Sanne) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:151]: I'm wondering if this should be clarified: technically these options are passed directly to the PG database server each time a new connection is initiated.

Comment (edeandrea) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:151]: @Sanne What if we said something like this?

```
Other unspecified properties to be passed through the Reactive SQL Client directly to the database when new connections are initiated.
```

Comment (Sanne) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:151]: yes sounds good! thanks

Comment (edeandrea) [reply extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java]: Done. Just pushed an update.

**Review by Sanne — APPROVED** [review]
Great, thanks!

**Pull request merged**